### PR TITLE
Initial Release of Sysdig Secure Integration

### PR DESCRIPTION
**Pull request by cole-labar**
## What does this PR do?

This is an initial release of a new integration for [Sysdig Secure](https://docs.sysdig.com/en/docs/sysdig-secure/). It captures events that are created by Sysdig's Rules. It includes:

A data stream for events from Sysdig rules.
Ingest pipeline for events data stream
Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files
Dashboard and visualizations of events.
Test for pipeline for event data stream.
System test cases for event data stream.
Documentation for users on how to configure Sysdig for this integration.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## What's Ready to Review

- [X] Data Stream & Mappings
- [X] Ingest Pipelines
- [X] Pipeline Tests
- [x] System (Integration) Tests
- [x] Visualizations
- [x] Documentation

## Related issues

- References [Sysdig Secure](https://github.com/elastic/integrations/issues/9564)

## Screenshots
Screenshot 2024-08-22 at 3 26 32 PM
Screenshot 2024-08-22 at 2 50 30 PM

### CODE REVIEW

File: packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+- set:
+    field: cloud.account.id
+    copy_from: sysdig.labels.cloudProvider.account.id
+    ignore_empty_value: true
+- set:
+    field: cloud.availability_zone
+    copy_from: sysdig.labels.gcp.availabilityZone
+    ignore_empty_value: true
+    if: ctx?.sysdig?.labels?.cloudProvider?.name != null && ctx?.sysdig?.labels?.cloudProvider?.name == "gcp"
+- script:
```
Comment (chemamartinez) [inline packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml:217]: Please review this script and the set of cloud fields in general, for some reason, cloud fields are not properly merged:
```
"cloud": {
    "account": {
        "id": "289645096542"
    },
    "availability_zone": "us-central1-c",
    "project": {
        "id": "289645096542",
        "name": "alliances-chronicle"
    },
    "provider": "gcp",
    "region": "us-central1"
},
"cloud.instance.id": "648229130641697246",
"cloud.instance.name": "gke-cluster-gcp-demo-san-default-pool-66250c41-vd1o",
"cloud.machine.type": "e2-standard-4",
```

**Review by chemamartinez — CHANGES_REQUESTED** [review]

File: packages/sysdig/validation.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+errors:
```
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/validation.yml]: Done!

File: packages/sysdig/kibana/dashboard/sysdig-1b3c9930-f0c2-46c8-9bce-47738b8bfa33.json
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/kibana/dashboard/sysdig-1b3c9930-f0c2-46c8-9bce-47738b8bfa33.json:1]: Updated! Controls have been added, Hide zero values has been disabled on the Alerts over Time, and Events have been renamed to Alerts to match the integrations focus

File: packages/sysdig/_dev/build/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,294 @@
+# Sysdig Integration
+This integration allows for the shipping of [Sysdig](https://sysdig.com/) alerts to Elastic for observability and organizational awareness. Alerts can then be analyzed by using either the dashboard included with the integration or via the creation of custom dashboards within Kibana.
+
+## Data Streams
+The Sysdig integration collects one type of data stream: logs.
```
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/_dev/build/docs/README.md]: Updated

File: packages/sysdig/_dev/build/docs/README.md [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+
+Sysdig alerts can contain a multitude of various fields pertaining to the type of activity on the host machine.
+
+{{ fields "alerts" }}
+
+**Example event**
+
+An example event for `alerts` looks as following:
+
+```json
```
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/_dev/build/docs/README.md]: Updated

File: packages/sysdig/changelog.yml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# newer versions go on top
+- version: "0.1.0"
+  changes:
+    - description: Initial draft of the Sysdig integration
```
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/changelog.yml]: Updated

File: packages/sysdig/data_stream/alerts/manifest.yml [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+        show_user: true
+      - name: data_stream.dataset
+        type: text
+        title: Dataset name
+        description: |
+          Dataset to write data to. Changing the dataset will send the data to a different index. You can't use `-` in the name of a dataset and only valid characters for [Elasticsearch index names](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html).
+        default: sysdig.alerts
+        required: true
+        show_user: true
+      - name: pipeline
```
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/manifest.yml]: Updated to remove this

File: packages/sysdig/_dev/build/docs/README.md [hunk 7]
```diff
@@ -0,0 +1,294 @@
+# Sysdig Integration
+This integration allows for the shipping of [Sysdig](https://sysdig.com/) alerts to Elastic for observability and organizational awareness. Alerts can then be analyzed by using either the dashboard included with the integration or via the creation of custom dashboards within Kibana.
+
+## Data Streams
+The Sysdig integration collects one type of data stream: logs.
+
+**Logs** The Logs data stream collected by the Sysdig integration is comprised of Sysdig Alerts. See more details about Sysdig Alerts in [Sysdig's Alerts Documentation](https://docs.sysdig.com/en/docs/sysdig-monitor/alerts/). A complete list of potential fields used by this integration can be found in the [Logs reference](#logs-reference)
```
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/_dev/build/docs/README.md]: Updated!

File: packages/sysdig/data_stream/alerts/manifest.yml [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+    mappings:
+      dynamic: false
+streams:
+  - input: http_endpoint
+    enabled: false
+    template_path: http.yml.hbs
+    title: HTTP input
+    description: Collect Sysdig alerts using input over HTTP
+    vars:
+      - name: method
```
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/manifest.yml]: Removed!

File: packages/sysdig/data_stream/alerts/agent/stream/http.yml.hbs [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+{{/if}}
+{{#if preserve_duplicate_custom_fields}}
+preserve_duplicate_custom_fields: true
+{{/if}}
+tags:
+{{#if preserve_original_event}}
+  - preserve_original_event
+{{/if}}
+{{#if preserve_duplicate_custom_fields}}
+  - preserve_duplicate_custom_fields
```
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/agent/stream/http.yml.hbs]: Updated

File: packages/sysdig/data_stream/alerts/agent/stream/http.yml.hbs [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+{{/if}}
+{{#if preserve_original_event}}
+preserve_original_event: true
+{{/if}}
+{{#if preserve_duplicate_custom_fields}}
+preserve_duplicate_custom_fields: true
+{{/if}}
+tags:
+{{#if preserve_original_event}}
+  - preserve_original_event
```
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/agent/stream/http.yml.hbs]: Updated

File: packages/sysdig/data_stream/alerts/manifest.yml [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+        show_user: false
+        description: |
+          Processors are used to reduce the number of fields in the exported event or to enhance the event with metadata. This executes in the agent before the logs are parsed. See [Processors](https://www.elastic.co/guide/en/beats/filebeat/current/filtering-and-enhancing-data.html) for details.
+      - name: tags
+        type: text
+        title: Tags
+        description: Tags to include in the published event
+        required: false
+        default:
+          - forwarded
```
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/manifest.yml:166]: Done!

File: packages/sysdig/data_stream/alerts/manifest.yml [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+        required: true
+        show_user: true
+        default: 9035
+      - name: url
+        type: text
+        title: URL
+        description: This options specific which URL path to accept requests on. Defaults to /.
+        required: false
+        show_user: true
+      - name: data_stream.dataset
```
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/manifest.yml]: Updated

File: packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml [hunk 13]
```diff
@@ -0,0 +1,295 @@
...
+    ignore_missing: true
+- set:
+    field: rule.category
+    copy_from: sysdig.content.ruleType
+    ignore_empty_value: true
+- set:
+    field: rule.ruleset
+    copy_from: sysdig.name
+    ignore_empty_value: true
+- set:
```
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml:93]: Not 100% sure why they were split out, but I do agree that nesting them under an `event` field would be better. I'll take care of that.

File: packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml]: Updated! These should be nesting appropriately now

File: packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml (hunk 13 above)
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml:93]: Updated

File: packages/sysdig/data_stream/alerts/manifest.yml [hunk 14]
```diff
@@ -0,0 +1,206 @@
...
+        default: false
+      - name: preserve_duplicate_custom_fields
+        required: true
+        show_user: true
+        title: Preserve duplicate custom fields
+        description: Preserve custom fields for all ECS mappings.
+        type: bool
+        multi: false
+        default: false
+      - name: prefix
```
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/manifest.yml]: Alright, I think I've pulled out the non-essential ones and left the ones that are referenced in the documentation. The `.hbs` file has also been updated.

File: packages/sysdig/data_stream/alerts/agent/stream/http.yml.hbs (hunk 9 above)
[the remark this answers is not in the document]
Comment (chemamartinez) [reply packages/sysdig/data_stream/alerts/agent/stream/http.yml.hbs]: The option is still in the manifest. Please remove it from there as well.

File: packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml (hunk 13 above)
[the remark this answers is not in the document]
Comment (chemamartinez) [reply packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml:93]: `SysdigEvent` is still defined in fields.yml.

File: packages/sysdig/data_stream/alerts/fields/fields.yml [hunk 15]
```diff
@@ -0,0 +1,212 @@
...
+  type: group
+  description: Incoming field parent
+  fields:
+    - name: category
+      type: text
+    - name: description
+      type: text
+    - name: type
+      type: text
+- name: host.name
```
Comment (chemamartinez) [inline packages/sysdig/data_stream/alerts/fields/fields.yml:187]: From this one down they look like ECS fields, so they don't need to be defined here.

File: packages/sysdig/data_stream/alerts/_dev/test/system/test-default-config.yml [hunk 16]
```diff
@@ -0,0 +1,14 @@
+service: sysdig-alerts-http
+service_notify_signal: SIGHUP
+input: http_endpoint
+vars: ~
+data_stream:
+  vars:
+    dataset: 'sysdig.alerts'
```
Comment (chemamartinez) [inline packages/sysdig/data_stream/alerts/_dev/test/system/test-default-config.yml:7]: `dataset` and `namespace` are no longer data stream options. You can remove them from here.

File: packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml [hunk 17]
```diff
@@ -0,0 +1,289 @@
+---
+description: Pipeline for mapping Sysdig fields to ECS / sysdig-namespaced fields
+processors:
+- set:
+    field: event.kind
+    value: 'alert'
+- json:
+    field: message
+    target_field: json
+    if: ctx?.message != null
```
Comment (chemamartinez) [inline packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml:10]:
```suggestion
    if: ctx.message != null
```

`ctx` is assumed that always exist, so can you replace every `ctx?` by just `ctx`?

File: packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml [hunk 18]
```diff
@@ -0,0 +1,289 @@
...
+    copy_from: sysdig.labels.kubernetes.pod.name
+    ignore_empty_value: true
+- set:
+    field: orchestrator.resource.parent.type
+    copy_from: sysdig.labels.kubernetes.workload.type
+    ignore_empty_value: true
+- set:
+    field: orchestrator.type
+    value: kubernetes
+    if: ctx?.sysdig?.labels?.kubernetes?.cluster?.name != null || ctx?.sysdig?.labels?.kubernetes?.namespace?.name != null || ctx?.sysdig?.labels?.kubernetes?.pod?.name != null || ctx?.sysdig?.labels?.kubernetes?.workload?.type != null
```
Comment (chemamartinez) [inline packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml:282]: Easier to read:
```suggestion
if: >-
  ctx?.sysdig?.labels?.kubernetes?.cluster?.name != null ||
  ctx?.sysdig?.labels?.kubernetes?.namespace?.name != null ||
  ctx?.sysdig?.labels?.kubernetes?.pod?.name != null ||
  ctx?.sysdig?.labels?.kubernetes?.workload?.type != null
```

File: packages/sysdig/data_stream/alerts/manifest.yml (hunk 14 above)
[the remark this answers is not in the document]
Comment (chemamartinez) [reply packages/sysdig/data_stream/alerts/manifest.yml]: I think the following options are still missing in the `http.yml.hbs` file but present in the manifest:
- url (needed)
- basic_auth
- username
- password
- hmac_header
- hmac_key
- hmac_prefix
- response_code
- response_body
- include_headers

I'd say most of them are not needed for our use case. The documentation is generic for the HTTP endpoint input, but we need to adjust to the Sysdig specifications when working with HTTP webhooks.

File: packages/sysdig/data_stream/alerts/agent/stream/http.yml.hbs (hunk 9 above)
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/agent/stream/http.yml.hbs]: Removed

File: packages/sysdig/data_stream/alerts/agent/stream/http.yml.hbs (hunk 10 above)
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/agent/stream/http.yml.hbs]: Done!

File: packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml (hunk 13 above)
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml:93]: I pulled them out now

File: packages/sysdig/data_stream/alerts/manifest.yml [hunk 19]
```diff
@@ -0,0 +1,151 @@
...
+        show_user: false
+      - name: ssl
+        type: yaml
+        title: TLS
+        description: Options for enabling TLS for the listening webhook endpoint. See the [documentation](https://www.elastic.co/guide/en/beats/filebeat/current/configuration-ssl.html) for a list of all options.
+        multi: false
+        required: false
+        show_user: false
+        default: |
+          enabled: false
```
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/manifest.yml:114]: Done!

File: packages/sysdig/data_stream/alerts/_dev/test/system/test-default-config.yml (hunk 16 above)
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/_dev/test/system/test-default-config.yml]: Updated

File: packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml (hunk 17 above)
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml]: Makes sense to me - updated

File: packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml (hunk 18 above)
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml]: Thanks for this! Updated

File: packages/sysdig/data_stream/alerts/manifest.yml (hunk 14 above)
[the remark this answers is not in the document]
Comment (cole-labar) [reply packages/sysdig/data_stream/alerts/manifest.yml]: This should now be updated to include only the relevant configs now

[the remark this answers is not in the document]
Comment (chemamartinez) [reply packages/sysdig/data_stream/alerts/manifest.yml]: I still think following changes are needed for consistency:
**manifest.yml**
- Remove basic_auth
- Remove username
- Remove password
- Add secret header
- Add secret value

**http.yml.hbs**
- Add URL

**Review by chemamartinez — COMMENTED** [review]
PR changes LGTM! Waiting for a manual test in a real scenario to approve it.

**Comment by cole-labar** [comment]
> PR changes LGTM! Waiting for a manual test in a real scenario to approve it.

Integration was connected to a Sysdig event forwarder for "live environment" testing and data was successfully intercepted!

**Review by chemamartinez — APPROVED** [review]

**Pull request merged**